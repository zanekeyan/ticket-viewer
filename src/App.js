import React, { Component } from "react";
import store from "./store";
import { BrowserRouter as Router, Route } from "react-router-dom";
import { Provider } from "react-redux";
import { fetchTickets} from './store/actions/TicketActions'
import Home from "./view/Home";
import './App.css';



class App extends Component{

    componentDidMount() {
        store.dispatch(fetchTickets());
    }
    render () {
        return (
            <Provider store={store}>
                <Router>
                    <Route exact path="/" component={Home} />
                </Router>
            </Provider>
        );
    }
}



export default App;
