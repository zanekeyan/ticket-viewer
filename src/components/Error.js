import React from 'react';
import bugIcon from '../bug.png';
import '../stylesheets/error.css'


const Error = props => {


    return (
        <div className="error-div">
            <h1> Error: {props.message} </h1>    
            <img
                src = {bugIcon}
                className="bug-icon"
                alt="car icon" />
        </div>
      );

}

export default Error;