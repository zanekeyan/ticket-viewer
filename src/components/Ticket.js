import React from 'react';
import PropTypes from 'prop-types';
import ticketicon from '../ticket.png'


const Ticket = props => {
    return (
        <div className="ticket-div" >
       <li className="list-group-item bg-dark text-white">
            <div className="ticketImage">
            <img
                src = {ticketicon}
                className="img-thumbnail"
                alt="car icon" />
            </div>
            <div className="ticketInfo">
            <p> subject: {props.subject} </p>
            <p> ticket_id: {props.id} </p> 
            <p> created_at: {props.created_at}</p>
            <p> submitter_id: {props.submitter_id} </p>
            <button
                type="button"
                className="btn btn-outline-light shadow-lg"
                onClick={() => { }} 
            > Ticket Details </button>
             
            </div>

        </li>
        </div>
      );     
}

Ticket.propTypes = {
    id: PropTypes.string,
    subject: PropTypes.string
};

export default Ticket;