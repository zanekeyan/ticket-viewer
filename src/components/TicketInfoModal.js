import React, { Component } from "react";
import Modal from "react-bootstrap-modal";

class TicketInfoModal extends Component {
  render() {
    return (
        <React.Fragment>
        <Modal
        {...this.props}
        animation="true"
        aria-labelledby="contained-modal-title-vcenter"
        centered="true"
        className="modalPopup">   
            <Modal.Header className="modal-header" closeButton>
                <Modal.Title className="modal-title">subject: {this.props.ticket.subject} </Modal.Title>
            </Modal.Header>
            <h3>Status: {this.props.ticket.status} </h3>
            <Modal.Body>
                <div className="container ">
                <div className=" detailContainer text-center ">
                    <p> description: {this.props.ticket.description} </p>
                    <div className="more-details-container text-left ">
                    <div className="row detail-row">
                    <p> submitter: {this.props.ticket.submitter_id} </p>
                    <p> requester: {this.props.ticket.requester_id} </p>
                    <p> group: {this.props.ticket.group_id} </p>
                    <p> created_at: {this.props.ticket.created_at} </p>
                    <p> updated_at: {this.props.ticket.updated_at} </p>
                    </div>
                    </div>
                </div>
                </div>
            </Modal.Body>
            </Modal>
        </React.Fragment>
    );
  }
}

export default TicketInfoModal;
