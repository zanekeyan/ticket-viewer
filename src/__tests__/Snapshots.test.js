import React from 'react';
import renderer from 'react-test-renderer'
import App from '../App';
import Error from '../components/Error';
import Ticket from '../components/Ticket'
import TicketList from '../components/TicketsList'
import TicketInfoModal from '../components/TicketInfoModal'
import {emptyTicket} from '../Utils.js/index'

const setUp = (Component  ) => {
    const component = renderer.create(<Component />);
    return component;
};

test('App snapshot test' , () => {
    const component = setUp(App)
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

test('Ticket snapshot test' , () => {
    const component = setUp(Ticket)
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

test('TicketList snapshot test' , () => {
    const component = setUp(TicketList)
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

test('Error snapshot test' , () => {
    const component = setUp(Error)
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

test('TicketInfoModal snapshot test' , () => {
    const component = renderer.create(<TicketInfoModal  ticket={emptyTicket} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})


