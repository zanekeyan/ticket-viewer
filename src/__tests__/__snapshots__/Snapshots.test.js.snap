// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test 1`] = `
<div
  className="tickets-list"
>
  <ul
    className="list-group  my-list-group bg-dark list-group-flush "
  />
</div>
`;

exports[`Error snapshot test 1`] = `
<div
  className="error-div"
>
  <h1>
     Error: 
     
  </h1>
  <img
    alt="car icon"
    className="bug-icon"
    src="bug.png"
  />
</div>
`;

exports[`Ticket snapshot test 1`] = `
<div
  className="ticket-div"
>
  <li
    className="list-group-item bg-dark text-white"
  >
    <div
      className="ticketImage"
    >
      <img
        alt="car icon"
        className="img-thumbnail"
        src="ticket.png"
      />
    </div>
    <div
      className="ticketInfo"
    >
      <p>
         subject: 
         
      </p>
      <p>
         ticket_id: 
         
      </p>
      <p>
         created_at: 
      </p>
      <button
        className="btn btn-outline-light shadow-lg"
        onClick={[Function]}
        type="button"
      >
         Ticket Details 
      </button>
    </div>
  </li>
</div>
`;

exports[`TicketInfoModal snapshot test 1`] = `null`;

exports[`TicketList snapshot test 1`] = `
<div
  className="tickets-list"
>
  <ul
    className="list-group  my-list-group bg-dark list-group-flush "
  />
</div>
`;
